.github/workflows/pre-commit-autoupdate.yml

```diff
@@ -0,0 +1,79 @@
+name: "Run pre-commit autoupdate"
+
+on:
+  workflow_call:
+    inputs:
+      target:
+        description: "The branch to issue the PR against"
+        type: "string"
+        default: "main"
+    secrets:
+      TOKEN:
+        description: "Your token"
+        required: true
+
+permissions:
+  contents: write
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  precommit-autoupdate:
+    name: "pre-commit autoupdate"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check out repository"
+        uses: actions/checkout@v4
```

Comment:
This checkout should be made from the target to prevent any conflicts from main and the target diverging.

Reply:
If I'm removing the target input, then I think this one is moot.